### PR TITLE
Enable importing passwords from Firefox

### PR DESCRIPTION
**Pull request by evq**
Auditors: @darkdh
required by https://github.com/brave/browser-laptop/pull/9433

### CODE REVIEW

File: atom/browser/importer/profile_writer.cc [hunk 1]
```diff
@@ -57,14 +62,8 @@ bool ProfileWriter::TemplateURLServiceIsLoaded() const {
...
-  if (importer_) {
-    base::DictionaryValue imported_form;
-    imported_form.SetString("signon_realm", form.signon_realm);
-    imported_form.SetString("username_value", form.username_value);
-    imported_form.SetString("password_value", form.password_value);
-
-    importer_->Emit("add-password-form", imported_form);
-  }
+    PasswordStoreFactory::GetForProfile(
+        ProfileManager::GetActiveUserProfile(), ServiceAccessType::EXPLICIT_ACCESS)->AddLogin(form);
```
Comment (darkdh) [inline atom/browser/importer/profile_writer.cc]: I think we should do null check of Profile*

File: brave/utility/importer/firefox_importer.cc [hunk 2]
```diff
@@ -87,4 +96,44 @@ void FirefoxImporter::ImportCookies() {
...
+      "WHERE type='login-saving' AND permission=2 AND expireType=0";
+
+
+  std::vector<autofill::PasswordForm> forms;
+  sql::Statement s(db.GetUniqueStatement(query));
+
+  while (s.Step() && !cancelled()) {
+    autofill::PasswordForm form;
+    base::string16 origin = s.ColumnString16(0);
+    form.origin = GURL(base::StringPiece16(origin));
```
Comment (darkdh) [inline brave/utility/importer/firefox_importer.cc]: you can directly use `GURL(s.ColumnString16(0));` here

File: brave/utility/importer/firefox_importer.cc [hunk 3]
```diff
@@ -87,4 +96,44 @@ void FirefoxImporter::ImportCookies() {
...
+    form.signon_realm = base::UTF16ToUTF8(origin) + '/';
```
Comment (darkdh) [inline brave/utility/importer/firefox_importer.cc]: use `form.origin.GetOrigin().spec()` would be better

File: patches/master_patch.patch [hunk 4]
```diff
@@ -907,10 +907,17 @@ index 7e05c4e04f8e61b418d373a73bd8901347df4c76..e7e6d9fa5e0d54c7755fe3a38eca6f70
...
-index 7f8d598bca0243e62f11d5d2ce50d8c25ba88764..d198d90e8872eb1d7a371aeaca6119c73a6cda1c 100644
+index 7f8d598bca0243e62f11d5d2ce50d8c25ba88764..72ea878a4e595a2761653dc108d2d9b448f3d965 100644
 --- a/chrome/utility/importer/firefox_importer.cc
 +++ b/chrome/utility/importer/firefox_importer.cc
-@@ -147,7 +147,6 @@ void FirefoxImporter::StartImport(const importer::SourceProfile& source_profile,
+@@ -140,14 +140,12 @@ void FirefoxImporter::StartImport(const importer::SourceProfile& source_profile,
+   if ((items & importer::PASSWORDS) && !cancelled()) {
+     bridge_->NotifyItemStarted(importer::PASSWORDS);
+     ImportPasswords();
+-    bridge_->NotifyItemEnded(importer::PASSWORDS);
```
Comment (bridiver) [inline patches/master_patch.patch]: why are we leaving out the notification for passwords?

Comment (bridiver) [inline patches/master_patch.patch]: I see now, we also do part of the import in `brave/utility/importer/firefox_importer.cc`
Is this something we can upstream so we don't have to patch in the future?

**Comment by bridiver** [comment]
++

**Pull request merged**